self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectsTestCase.java

```diff
@@ -81,6 +83,37 @@ public void projectOwnedBy() {
                 Matchers.iterableWithSize(1));
     }
 
+    /**
+     * Check if the right project is returned.
+     */
+    @Test
+    public void getProjectById() {
+        final Storage storage = new InMemory();
+        ProjectManager projectManager = storage
+                .projectManagers().pick("github");
+        InMemoryProjects projects = (InMemoryProjects)
+                storage.projects();
+        projects.register(mock(Repo.class), projectManager);
+        projects.register(mock(Repo.class), projectManager);
+        Project project = projects.getProjectById(1);
+        assertThat(project.projectId(), equalTo(1));
+    }
+
+    /**
+     * Check if no project is returned when id is not found.
+     */
+    @Test
+    public void getProjectByNotFoundId() {
+        final Storage storage = new InMemory();
+        ProjectManager projectManager = storage
+                .projectManagers().pick("github");
+        InMemoryProjects projects = (InMemoryProjects)
+                storage.projects();
+        projects.register(mock(Repo.class), projectManager);
```

Comment:
@BogdanAdrian11 you don't need to register any project for this test. Just a ``getById`` with ``-1`` is logically enough.